tools/TestDBGenerator/Orchestrations.cs

```diff
@@ -0,0 +1,211 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace TestDBGenerator;
+
+using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using DurableTask.Core;
+using DurableTask.Core.History;
+using DurableTask.SqlServer;
+
+static class Orchestrations
+{
+    static readonly ManualResetEvent ActivityBlockedEvent = new(initialState: false);
+    
+    static bool PauseActivityExecution { get; set; } = false;
+
+    public static async Task GenerateRuntimeDataAsync(SqlOrchestrationService service)
```

Comment:
this method needs to create a very specific end state for the workflows it schedules, which isn't immediately obvious to me unless I'm cross checking with the `UpgradeTests.cs` checks.

Any chance we could document these runtime end states in here? Would make this method;s purpose easier to understand.

Reply:
Do the existing comments inside the method not clarify what the expected outcomes are? Maybe you can clarify what you mean by "end state"?

Reply:
By end state I meant the checks that we will be making for these orchestrations. For example, here we say that "Test2" is scheduled with a delay, but that doesn't make it immediately obvious that the reason for this is that we want it to remain in the "Running" state until the DB is restored. It makes sense in retrospective, but when I first read through this file (and before I got to the tests) I wasn't sure what we were trying to accomplish with this data generation step. I think the comments in describe more explicitly the state of all these orchestrators when the taskhub-worker is shut down.

_But_ now that I've re-read the comments in this file, I find my feedback here to be too nitpicky so I'll retract it :) .
At most,, I would just suggest updating the comment here for "Test2" to make it clearer that we expect the shutdown worker to shut down before it gets a chance to run.

Reply:
I added a bit more commentary as part of https://github.com/microsoft/durabletask-mssql/pull/130/commits/c164131f63c135b0dfa4121172234d5385eaf179, which will hopefully make the intent more clear.